src/engine/tt_hash_table.h

```diff
@@ -14,20 +15,87 @@ enum TtHashFlag : uint8_t {
     TtHashBeta = 2,
 };
 
+/*
+ * Non-key members are encoded for compression. From lower bits to higher, entries are:
+ * depth: 7 bits, i.e. 0000 ... 0000 0000 0111 1111
+ * flag: 3 bits, i.e. 0000 ... 0000 0011 1000 0000
+ * score: 17 bits (see board_defs.h)
+ * generation: 8 bits
+ * move: 26 bits (see move_types.h)
+ * */
+
+class TtHashEntryData {
+public:
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wshift-count-overflow"
+    explicit TtHashEntryData(uint64_t depth = 0, uint64_t flag = TtHashExact, int64_t score = 0, uint64_t generation = 0, uint64_t move = {})
```

Comment:
Pretty sure that this doesn't work with the signed value (score). We need to store the signedness somewhere - hence the overflow warning.

There's a few ways to do it, eg:

1. Have a signed bit in our encoded data (1 if score is negative and then store score as abs value).

2. Offset the absolute score and do the opposite when decoding.